### DISCUSSION TITLE
"Lock file error" with no further details

### DISCUSSION THREAD

**Discussion by l0b0** [renovatebot/renovate#29880] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

gitlab.com, Renovate v37.414.1

### Please tell us more about your question or problem

Timeline, oldest first:

1. [Last passing job at 2024-06-24T02:03:05.550Z](https://gitlab.com/engmark/renovate/-/jobs/7167340238) for the [mypy-exercises repo](https://gitlab.com/engmark/mypy-exercises/)
1. [First job with an error at 2024-06-25T02:03:06.656Z](https://gitlab.com/engmark/renovate/-/jobs/7177080635)
1. [Resulting change from the failing job](https://gitlab.com/engmark/mypy-exercises/-/merge_requests/557)
2. Checked "If you want to rebase/retry this MR, check this box"
3. [Failing retry job at 2024-06-27T02:02:22.596Z](https://gitlab.com/engmark/renovate/-/jobs/7197993167)
4. MR updated at 2024-06-27T02:06:45.702Z without any changes to the diff.

Crucially, [there were no changes to the mypy-exercises repo in this period](https://gitlab.com/engmark/mypy-exercises/-/merge_requests?scope=all&state=merged) and *both* Renovate jobs ran with the same commit (as seen by "Commit `73bb6ff0`" in both jobs), that is, the same [renovate-runner](https://gitlab.com/engmark/renovate/-/merge_requests/284) and Renovate versions (unless someone moved the renovate-runner v17.263.1 tag). So I'm a bit stumped for what happened between these two runs, except for the upstream mypy version changing. [The error report](https://gitlab.com/engmark/mypy-exercises/-/merge_requests/557#note_1967604404) doesn't tell me anything more.

### Logs (if relevant)

<details><summary>Logs</summary>

First failing job:

```
 INFO: Repository started (repository=engmark/mypy-exercises)
       "renovateVersion": "37.415.0"
 INFO: Dependency extraction complete (repository=engmark/mypy-exercises, baseBranch=main)
       "stats": {
         "managers": {
           "gitlabci": {"fileCount": 1, "depCount": 5},
           "pep621": {"fileCount": 1, "depCount": 1},
           "poetry": {"fileCount": 1, "depCount": 6},
           "regex": {"fileCount": 1, "depCount": 1}
         },
         "total": {"fileCount": 4, "depCount": 13}
       }
 WARN: No tool releases found. (repository=engmark/mypy-exercises, branch=renovate/mypy)
       "toolConfig": {
         "datasource": "github-releases",
         "packageName": "containerbase/python-prebuild",
         "versioning": "python"
       }
 INFO: Lock file error - aborting (repository=engmark/mypy-exercises)
 INFO: Repository finished (repository=engmark/mypy-exercises)
       "cloned": true,
       "durationMs": 20868
```

Retry job:

```
 INFO: Repository started (repository=engmark/mypy-exercises)
       "renovateVersion": "37.419.1"
 INFO: Dependency extraction complete (repository=engmark/mypy-exercises, baseBranch=main)
       "stats": {
         "managers": {
           "gitlabci": {"fileCount": 1, "depCount": 5},
           "pep621": {"fileCount": 1, "depCount": 1},
           "poetry": {"fileCount": 1, "depCount": 6},
           "regex": {"fileCount": 1, "depCount": 1}
         },
         "total": {"fileCount": 4, "depCount": 13}
       }
 WARN: No tool releases found. (repository=engmark/mypy-exercises, branch=renovate/mypy)
       "toolConfig": {
         "datasource": "github-releases",
         "packageName": "containerbase/python-prebuild",
         "versioning": "python"
       }
 INFO: Branch updated (repository=engmark/mypy-exercises, branch=renovate/mypy)
       "commitSha": "d00c819f38a4be40075cc93ea323ceae3b9a8009"
 WARN: Package lookup failures (repository=engmark/mypy-exercises, branch=renovate/mypy)
       "warnings": ["Failed to look up github-releases package containerbase/python-prebuild"],
       "files": ["pyproject.toml"]
 INFO: PR updated (repository=engmark/mypy-exercises, branch=renovate/mypy)
       "pr": 557,
       "prTitle": "chore(deps): update dependency mypy to v1.10.1"
 WARN: artifactErrors (repository=engmark/mypy-exercises, branch=renovate/mypy)
       "artifactErrors": [{"lockFile": "poetry.lock", "stderr": "undefined\nundefined"}]
 WARN: No tool releases found. (repository=engmark/mypy-exercises, branch=renovate/pytest-8.x-lockfile)
       "toolConfig": {
         "datasource": "github-releases",
         "packageName": "containerbase/python-prebuild",
         "versioning": "python"
       }
 WARN: Detected empty commit - aborting git push (repository=engmark/mypy-exercises, branch=renovate/pytest-8.x-lockfile)
       "commitRes": {
         "author": null,
         "branch": "",
         "commit": "",
         "root": false,
         "summary": {"changes": 0, "insertions": 0, "deletions": 0}
       }
 INFO: Repository finished (repository=engmark/mypy-exercises)
       "cloned": true,
       "durationMs": 27049
```

</details>

**Comment by RahulGautamSingh** [comment]
I think it failed because the lookup failed for the python-prebuild from containerbase. We need python installed with poetry so that lockfile can be generated, hence the lockfile error.

Maybe its a temporary connection issue which caused the lookup error. If it persists you can change the log level to debug for detailed logs.

**Reply by viceice** [reply]
I fixed a containerbase bug, which prevented download retry 

https://github.com/containerbase/base/releases/tag/10.11.8

**Reply by l0b0** [reply]
I did ask it to retry once; I've added info about the retry above. But I also realised that the GitHub API token had expired, which seems to have been the actual underlying issue. That said:

- The log output does not explicitly call out this issue. If at all possible it would be good if messages like "Failed to look up github-releases package containerbase/python-prebuild" would point out *why* it failed, ideally with enough info to realise that the GitHub token might have expired or not be configured.
- The [renovate-runner docs](https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/README.md?ref_type=heads#configure-cicd-variables) should probably be more explicit about the GitHub token being *necessary* for updating packages on GitHub.

**Reply by RahulGautamSingh** [reply]
1. You will get detailed logs if you set LOG_LEVEL to debug, by default it is set to info.
2. I think the current info seems fine ie. it mentions that you need a Github token to fetch changelogs & lookp dependency that use github tags. What more would you like it to say?

**Reply by l0b0** [reply]
> 1. You will get detailed logs if you set LOG_LEVEL to debug, by default it is set to info.

This is an error. Ideally the error message should include enough information to work out what has happened and how to correct it. Mentioning the rate limiting would be a good start, and mentioning that the GitHub token might be invalid/expired would be a huge improvement.
 
> 2. I think the current info seems fine ie. it mentions that you need a Github token to fetch changelogs & lookp dependency that use github tags. What more would you like it to say?

It would be great if it used more forceful language. The way it's currently phrased it's not clear just how limited the GitHub API rates are when unauthenticated, or how Renovate automatically backs off and retries (presumably it doesn't), or that Renovate doesn't have any other way to get at the required info (tags) to perform upgrades. It also puts changelog retrieval ahead of the much more important info about tags. So maybe instead of 

"It is also recommended to configure a GitHub.com Personal Access Token (minimum scopes) as `GITHUB_COM_TOKEN` so that your bot can make authenticated requests to github.com for Changelog retrieval as well as for any dependency that uses GitHub tags. Without such a token, github.com's API will rate limit requests and make such lookups unreliable."

, maybe something like (frontloads warning, doesn't mention changelogs):

"To reliably update dependencies which use GitHub tags it is recommended to configure a GitHub.com Personal Access Token (minimum scopes) as `GITHUB_COM_TOKEN`. GitHub rate limits API requests, and any such error will result in Renovate skipping the relevant dependency."

**Reply by rarkins** [reply]
PRs welcome for any of the above

**Reply by l0b0** [reply]
Unfortunately this project requires that I assign copyright over to WhiteSource Software, which I always assume means the project will eventually be closed source.

**Reply by rarkins** [reply]
You're welcome to keep using the software itself, but I don't think it's fair for you to continue consuming our time with technical support or feature requests in this repo while being philosophically opposed to contributing back in any way, including even improving the docs for other users.
